{"name": "Добавление товара в корзину", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.WebDriverException: Message:", "trace": "browser = <selenium.webdriver.remote.webdriver.WebDriver (session=\"0b1f28863b99867892897d97ac3e525e\")>, base_url = 'http://10.0.2.15:8081'\n\n    @allure.title(\"Добавление товара в корзину\")\n    def test_add_to_cart(browser, base_url):\n        home_page = HomePage(browser)\n        home_page.open(base_url)\n>       home_page.scroll_down()\n\ntests/test_start.py:60: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npage_objects/home_page.py:17: in scroll_down\n    self.browser.execute_script(\"window.scrollTo(0, document.body.scrollHeight);\")\nvenv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:407: in execute_script\n    return self.execute(command, {\"script\": script, \"args\": converted_args})[\"value\"]\nvenv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py:347: in execute\n    self.error_handler.check_response(response)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.remote.errorhandler.ErrorHandler object at 0x711a274371a0>, response = {'status': [13, 'unknown error'], 'value': ''}\n\n    def check_response(self, response: Dict[str, Any]) -> None:\n        \"\"\"Checks that a JSON response from the WebDriver does not have an\n        error.\n    \n        :Args:\n         - response - The JSON response from the WebDriver server as a dictionary\n           object.\n    \n        :Raises: If the response contains an error message.\n        \"\"\"\n        status = response.get(\"status\", None)\n        if not status or status == ErrorCode.SUCCESS:\n            return\n        value = None\n        message = response.get(\"message\", \"\")\n        screen: str = response.get(\"screen\", \"\")\n        stacktrace = None\n        if isinstance(status, int):\n            value_json = response.get(\"value\", None)\n            if value_json and isinstance(value_json, str):\n                import json\n    \n                try:\n                    value = json.loads(value_json)\n                    if len(value) == 1:\n                        value = value[\"value\"]\n                    status = value.get(\"error\", None)\n                    if not status:\n                        status = value.get(\"status\", ErrorCode.UNKNOWN_ERROR)\n                        message = value.get(\"value\") or value.get(\"message\")\n                        if not isinstance(message, str):\n                            value = message\n                            message = message.get(\"message\")\n                    else:\n                        message = value.get(\"message\", None)\n                except ValueError:\n                    pass\n    \n        exception_class: Type[WebDriverException]\n        e = ErrorCode()\n        error_codes = [item for item in dir(e) if not item.startswith(\"__\")]\n        for error_code in error_codes:\n            error_info = getattr(ErrorCode, error_code)\n            if isinstance(error_info, list) and status in error_info:\n                exception_class = getattr(ExceptionMapping, error_code, WebDriverException)\n                break\n        else:\n            exception_class = WebDriverException\n    \n        if not value:\n            value = response[\"value\"]\n        if isinstance(value, str):\n>           raise exception_class(value)\nE           selenium.common.exceptions.WebDriverException: Message:\n\nvenv/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py:193: WebDriverException"}, "steps": [{"name": "Открываю http://10.0.2.15:8081", "status": "passed", "start": 1725129375897, "stop": 1725129380086}, {"name": "Скроллю страницу вниз", "status": "broken", "statusDetails": {"message": "selenium.common.exceptions.WebDriverException: Message: \n\n", "trace": "  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/allure_commons/_allure.py\", line 192, in impl\n    return func(*a, **kw)\n           ^^^^^^^^^^^^^^\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/page_objects/home_page.py\", line 17, in scroll_down\n    self.browser.execute_script(\"window.scrollTo(0, document.body.scrollHeight);\")\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 407, in execute_script\n    return self.execute(command, {\"script\": script, \"args\": converted_args})[\"value\"]\n           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/selenium/webdriver/remote/webdriver.py\", line 347, in execute\n    self.error_handler.check_response(response)\n  File \"/home/fuenukkusu2501/PycharmProjects/OTUS_selenium_UI/venv/lib/python3.12/site-packages/selenium/webdriver/remote/errorhandler.py\", line 193, in check_response\n    raise exception_class(value)\n"}, "start": 1725129380088, "stop": 1725129470569}], "start": 1725129375897, "stop": 1725129470581, "uuid": "0eca3450-282b-4e46-ac9e-1a9d45490cd0", "historyId": "2987328dc021ab097458a9c10fd9e829", "testCaseId": "2987328dc021ab097458a9c10fd9e829", "fullName": "tests.test_start#test_add_to_cart", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_start"}, {"name": "host", "value": "fuenukkusu2501-VirtualBox"}, {"name": "thread", "value": "24912-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_start"}]}